test/e2e/framework/fixture.go

```diff
@@ -271,89 +271,187 @@ func NewWorkspaceWithWorkloads(t *testing.T, server RunningServer, orgClusterNam
+	for _, resource := range sf.ResourcesToSync.List() {
 		pluginArgs = append(pluginArgs, "--resources", resource)
 	}
 	syncerYAML := RunKcpCliPlugin(t, kubeconfigPath, pluginArgs)
 
-	// Create a fake server from a workspace that is a peer to the current workspace
-	// TODO(marun) Only use a fake workload server if a pcluster is not configured
-	parentClusterName, ok := cfg.WorkspaceClusterName.Parent()
-	require.True(t, ok, "%s does not have a parent", cfg.WorkspaceClusterName)
-	downstreamServer := NewFakeWorkloadServer(t, cfg.UpstreamServer, parentClusterName)
-	downstreamConfig := downstreamServer.DefaultConfig(t)
+	var downstreamConfig *rest.Config
+	var downstreamKubeconfigPath string
+	if useDeployedSyncer {
+		// The syncer will target the pcluster identified by `--pcluster-kubeconfig`.
+		downstreamKubeconfigPath = TestConfig.PClusterKubeconfig()
+		fs, err := os.Stat(downstreamKubeconfigPath)
+		require.NoError(t, err)
+		require.NotZero(t, fs.Size(), "%s points to an empty file", downstreamKubeconfigPath)
+		rawConfig, err := clientcmd.LoadFromFile(downstreamKubeconfigPath)
+		require.NoError(t, err, "failed to load pcluster kubeconfig")
+		config := clientcmd.NewNonInteractiveClientConfig(*rawConfig, rawConfig.CurrentContext, nil, nil)
+		downstreamConfig, err = config.ClientConfig()
+		require.NoError(t, err)
+	} else {
+		// The syncer will target a logical cluster that is a peer to the current workspace. A
+		// logical server provides as a lightweight approximation of a pcluster for tests that
+		// don't need to validate running workloads or interaction with kube controllers.
+		parentClusterName, ok := sf.WorkspaceClusterName.Parent()
+		require.True(t, ok, "%s does not have a parent", sf.WorkspaceClusterName)
+		downstreamServer := NewFakeWorkloadServer(t, sf.UpstreamServer, parentClusterName)
+		downstreamConfig = downstreamServer.DefaultConfig(t)
+		downstreamKubeconfigPath = downstreamServer.KubeconfigPath()
+	}
+
+	if sf.InstallCRDs != nil {
+		// Attempt crd installation to ensure the downstream server has an api surface
+		// compatible with the test.
+		sf.InstallCRDs(downstreamConfig, !useDeployedSyncer)
+	}
 
 	// Apply the yaml output from the plugin to the downstream server
-	KubectlApply(t, downstreamServer.KubeconfigPath(), syncerYAML)
+	KubectlApply(t, downstreamKubeconfigPath, syncerYAML)
 
 	// Extract the configuration for an in-process syncer from the resources that were
 	// applied to the downstream server. This maximizes the parity between the
 	// configuration of a deployed and in-process syncer.
-	syncerNamespace := workloadcliplugin.GetSyncerID(cfg.WorkspaceClusterName.String(), cfg.WorkloadClusterName)
+	syncerNamespace := workloadcliplugin.GetSyncerID(sf.WorkspaceClusterName.String(), sf.WorkloadClusterName)
 	syncerConfig := syncerConfigFromCluster(t, downstreamConfig, syncerNamespace)
 
+	ctx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
Should we pass ctx in as an arg to Start()?

Reply:
I don't think there's a need to manage context for most syncer, it can be an internal detail of the fixture. It's not like the use of context in a request chain, the only requirement is that the fixture gets cleaned up at the end of the test so it's not necessary to pass a single context around everywhere.